telosnetwork/telos-wallet
# new EVMAuthenticator to support Brave Wallet

# Fixes #644

## Description
Adding support for the brave wallet when is detected (only on desktop)

## Test scenarios
- use Chrome and go to [this link](https://deploy-preview-683--wallet-staging.netlify.app/)
  - as the screenshot shows, you should not see the Brave Wallet button 
- use Brave, disable your wallet (config -> web3 -> change default wallets to Extensions)
- go to the [same link](https://deploy-preview-683--wallet-staging.netlify.app/)
  - you should see the Brave Wallet button just after Telos Wallet
- Press the Brave Wallet button
  - you should get an error message like in the screenshot
- go and enable your wallet again (see last screenshot)
- try to log in once more time ([same link](https://deploy-preview-683--wallet-staging.netlify.app/))
  - you should be able to log in and perform any transaction allowed by the site.

## Screenshots

- Chrome Browser:
![Captura desde 2023-11-20 22-17-19](https://github.com/telosnetwork/telos-wallet/assets/4420760/222cc3c2-01ac-4c01-8470-387bec18dd02)

- Brave Browser:
![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/02a3a2d9-7b6f-49ed-b3b9-7cdd45d55cd9)

- Case when you are using the Brave Browser but your wallet is not enabled:
![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/044314ed-7833-459b-a766-2f0f229a3e5b)

- Brave configuration to disable/enable brave wallet
![image](https://github.com/telosnetwork/telos-wallet/assets/4420760/fbc11e2f-4e6c-45d1-967c-fdd668a66b92)


## Checklist:
